File: VB/Mvc_DbStorage_Sample_VB/DAL/ReportEntity.vb
Imports DevExpress.Xpo

Namespace Services.DAL
	<DeferredDeletion(False)>
	Public Class ReportEntity
		Inherits XPCustomObject

		Private url_Renamed As String
		Private name As String
		Private layout_Renamed() As Byte

		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub

		<Key>
		Public Property Url() As String
			Get
				Return url_Renamed
			End Get
			Set(ByVal value As String)
				SetPropertyValue("Url", url_Renamed, value)
			End Set
		End Property

		Public Property Layout() As Byte()
			Get
				Return layout_Renamed
			End Get
			Set(ByVal value As Byte())
				SetPropertyValue("Layout", layout_Renamed, value)
			End Set
		End Property
	End Class
End Namespace

File: CS/Mvc_DbStorage_Sample/DAL/ReportEntity.cs
using DevExpress.Xpo;

namespace Mvc_DbStorage_Sample.Services.DAL {
    [DeferredDeletion(false)]
    public class ReportEntity : XPCustomObject {
        string url;
        string name;
        byte[] layout;

        public ReportEntity(Session session)
            : base(session) {
        }

        [Key]
        public string Url {
            get { return url; }
            set { SetPropertyValue("Url", ref url, value); }
        }

        public byte[] Layout {
            get { return layout; }
            set { SetPropertyValue("Layout", ref layout, value); }
        }
    }
}
